ksql-engine/src/test/java/io/confluent/ksql/KsqlEngineTest.java

```diff
@@ -79,6 +80,30 @@ public void shouldFailWhenSyntaxIsInvalid() throws Exception {
     ksqlEngine.createQueries("blah;");
   }
 
+  @Test
+  public void shouldUpdateReferentialIntegrityTableCorrectly() throws Exception {
+    ksqlEngine.createQueries("create table bar as select * from test2;" +
+                                   "create table foo as select * from test2;");
+    MetaStore metaStore = ksqlEngine.getMetaStore();
+    assertThat(metaStore.getQueriesWithSource("TEST2"),
+               equalTo(Utils.mkSet("CTAS_BAR", "CTAS_FOO")));
+    assertThat(metaStore.getQueriesWithSink("BAR"), equalTo(Utils.mkSet("CTAS_BAR")));
+    assertThat(metaStore.getQueriesWithSink("FOO"), equalTo(Utils.mkSet("CTAS_FOO")));
+  }
+
+  @Test
+  public void shouldFailIfRererentialIntegrityIsViolated() {
+    try {
+      ksqlEngine.createQueries("create table bar as select * from test2;" +
+                               "create table foo as select * from test2;");
+      ksqlEngine.createQueries("drop table foo;");
+      Assert.fail();
+    } catch (Exception e) {
+      assertThat(e.getMessage(), equalTo(
```

Comment:
Would be better to assert that the exception is of type `KsqlReferentialIntegrityException` rather than trying to match the exception text.